conftest.py

```diff
@@ -87,6 +87,15 @@ def skipif(items, whole_module=False):
         if i == 'chkpnt' and Revolver is NoopRevolver:
             skipit = "pyrevolve not installed"
             break
+        if i == 'petsc':
+            petsc_dir = os.environ.get('PETSC_DIR', "")
```

Comment:
nitpicking: you may want to add some utility functions such as `get_petsc_env` which returns `(dir, arch)`? for example placed inside `devito/petsc/utils.py` or wherever you deem reasonable